fastcrypto-zkp/src/conversions.rs

```diff
@@ -532,38 +579,65 @@ pub(crate) mod tests {
 
     // Affine point roundtrips
 
+    fn bls_g1_affine_infinity() -> BlsG1Affine {
+        let res = BlsG1Affine::zero();
+        debug_assert_eq!(res, BlsG1Affine::new(Fq::zero(), Fq::one(), true));
+        res
+    }
+
     pub(crate) fn arb_bls_g1_affine() -> impl Strategy<Value = BlsG1Affine> {
         // slow, but good enough for tests
-        arb_bls_fr().prop_map(|s| {
-            BlsG1Affine::prime_subgroup_generator()
-                .mul(s.into_repr())
-                .into_affine()
+        (arb_bls_fr(), any::<f32>()).prop_map(|(s, maybe_infinity)| {
+            if maybe_infinity < 0.1 {
+                // 10% chance of being the point at infinity
```

Comment:
Can you please explain or provide a reference for this 10% probability?

Reply:
Good catch, this is a clunky and erroneous attempt at making this a minority case. The reason it kinda-sorta worked experimentally is that proptest divides floats in 7 strategies: https://docs.rs/proptest/1.0.0/proptest/num/f32/constant.ANY.html
I've refactored this (and other instances) to actually reflect what I intended to do.